pageserver/src/tenant.rs

```diff
@@ -1269,6 +1281,18 @@ impl Tenant {
             }
         };
 
+        if let Some(remote_client) = loaded_timeline.remote_client.as_ref() {
+            // failing the await for uploads to complete will mean that the tenant is
+            // non-relocatable, however with erroring out it is clear to control plane that the
+            // request needs to be retried, and next time after restart they will get 409.
```

Comment:
```suggestion
            // Wait for the upload of the 'index_part.json` file to finish, so that when we return
            // Ok, the timeline is durable in remote storage.
```